internal/pkg/graph/graph.go

```diff
@@ -401,19 +369,21 @@ func (t *graphTraversal[V]) execute(ctx context.Context, lg *LabeledGraph[V]) er
 		for _, vertex := range vertices {
 			vertex := vertex
 			// Delay processing this vertex if any of its dependent vertices are yet to be processed.
```

Comment:
Looking at the code, it seems like this is saying that if any of the "previous" vertices are yet to be processed, we delay the processing of "this" vertex. If this is correct, then nitnitnitnitnit:

```suggestion
			// If any of the vertices that should be visited before this vertex are yet to be processed, we delay processing it.
```